beets/util/artresizer.py

```diff
@@ -84,7 +88,7 @@ class IMBackend(LocalBackend):
     _legacy = None
 
     @classmethod
-    def version(cls):
+    def version(cls) -> Optional[Union[object, Tuple[int, int, int]]]:
```

Comment:
This may be out of scope for a typing PR, but I think this annotation is a bit much; if anything it highlights that the version logic is a bit hard to work with.

I'd suggest
* Some other way to handle "no version available" instead of the raw object instance
  * `None` would be way more clear
  * Could allow for that by changing the check below to `not hasattr(...)` instead of `... is None`
* Replace the "raw tuple" with a `NamedTuple`

All in all this would simplify those annotations by a lot:

```py
class Version(NamedTuple):
  major: int
  minor: int
  patch: int


class IMBacked(LocalBackend):
  @classmethod
  def version(cls) -> Optional[Version]:
    ...
```

Reply:
I think it is out of scope. I don't think I should be changing the logic of the methods I'm typing in this PR. If you feel differently then we can put changes in motion but I feel like there should be a different PR for those. There's a great deal of places where things could be changed or made simpler, but there's only so many changes before things become unmanageable.